pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -18,99 +18,287 @@ package volumebinding
+	pvutil "k8s.io/kubernetes/pkg/controller/volume/persistentvolume/util"
 	"k8s.io/kubernetes/pkg/controller/volume/scheduling"
+	"k8s.io/kubernetes/pkg/scheduler/apis/config"
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	"k8s.io/utils/pointer"
 )
 
-func TestVolumeBinding(t *testing.T) {
-	findErr := fmt.Errorf("find err")
-	volState := v1.PodSpec{
-		Volumes: []v1.Volume{
-			{
-				VolumeSource: v1.VolumeSource{
-					PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{},
+var (
+	immediate            = storagev1.VolumeBindingImmediate
+	waitForFirstConsumer = storagev1.VolumeBindingWaitForFirstConsumer
+	immediateSC          = &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "immediate-sc",
+		},
+		VolumeBindingMode: &immediate,
+	}
+	waitSC = &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "wait-sc",
+		},
+		VolumeBindingMode: &waitForFirstConsumer,
+	}
+)
+
+func makePV(name string) *v1.PersistentVolume {
+	return &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+func addPVNodeAffinity(pv *v1.PersistentVolume, volumeNodeAffinity *v1.VolumeNodeAffinity) *v1.PersistentVolume {
+	pv.Spec.NodeAffinity = volumeNodeAffinity
+	return pv
+}
+
+func makePVC(name string, boundPVName string, storageClassName string) *v1.PersistentVolumeClaim {
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: v1.NamespaceDefault,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			StorageClassName: pointer.StringPtr(storageClassName),
+		},
+	}
+	if boundPVName != "" {
+		pvc.Spec.VolumeName = boundPVName
+		metav1.SetMetaDataAnnotation(&pvc.ObjectMeta, pvutil.AnnBindCompleted, "true")
+	}
+	return pvc
+}
+
+func makePod(name string, pvcNames []string) *v1.Pod {
+	p := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: v1.NamespaceDefault,
+		},
+	}
+	p.Spec.Volumes = make([]v1.Volume, 0)
+	for _, pvcName := range pvcNames {
+		p.Spec.Volumes = append(p.Spec.Volumes, v1.Volume{
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvcName,
 				},
 			},
-		},
+		})
 	}
+	return p
+}
+
+func TestVolumeBinding(t *testing.T) {
 	table := []struct {
-		name               string
-		pod                *v1.Pod
-		node               *v1.Node
-		volumeBinderConfig *scheduling.FakeVolumeBinderConfig
-		wantStatus         *framework.Status
+		name                    string
+		pod                     *v1.Pod
+		node                    *v1.Node
+		pvcs                    []*v1.PersistentVolumeClaim
+		pvs                     []*v1.PersistentVolume
+		wantPreFilterStatus     *framework.Status
+		wantStateAfterPreFilter *stateData
+		wantFilterStatus        *framework.Status
 	}{
 		{
-			name:       "nothing",
-			pod:        &v1.Pod{},
-			node:       &v1.Node{},
-			wantStatus: nil,
+			name: "pod has not pvcs",
```

Comment:
renamed `nothing` to `pod has not pvcs`, the pod can be skipped if it does not use PVCs